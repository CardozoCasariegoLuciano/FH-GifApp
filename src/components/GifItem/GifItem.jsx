import React from "react";

const GifItem = ({gif}) => {
  return (
    <div className="card">
      <img src={gif.url} alt={gif.title}/>
      <p>{gif.title}</p>
    </div>
  );
};

export default GifItem;
