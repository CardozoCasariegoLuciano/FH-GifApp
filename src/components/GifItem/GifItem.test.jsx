import { render, screen } from "@testing-library/react";
import GifItem from "./GifItem";

describe("GifItem testSuit", () => {
  const title = "Test title";
  const url = "http://www.google/";

  test("Debe de hacer mach con el snapshot", async () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el URL y ALT indicados", async () => {
    render(<GifItem title={title} url={url} />);
    //screen.debug();
    const {src, alt} = screen.getByRole("img")

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe de mostrar el titulo en el componente", ()=>{
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy()
  })
});
