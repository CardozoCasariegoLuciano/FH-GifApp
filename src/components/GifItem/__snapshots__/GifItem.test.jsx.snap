// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem testSuit Debe de hacer mach con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test title"
      src="http://www.google/"
    />
    <p>
      Test title
    </p>
  </div>
</div>
`;
